src/gc/gcinterface.d

```diff
@@ -0,0 +1,194 @@
+/**
+ * Contains the internal GC interface.
+ *
+ * Copyright: Copyright Digital Mars 2016.
+ * License:   $(WEB www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+ * Authors:   Walter Bright, Sean Kelly, Jeremy DeHaan
+ */
+
+ /*          Copyright Digital Mars 2016.
+ * Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ */
+module gc.gcinterface;
+import gc.stats;
+
+static import core.memory;
+alias BlkAttr = core.memory.GC.BlkAttr;
+alias BlkInfo = core.memory.GC.BlkInfo;
+
+alias RootIterator = int delegate(scope int delegate(ref Root) nothrow dg);
+alias RangeIterator = int delegate(scope int delegate(ref Range) nothrow dg);
+
+
+struct Root
+{
+    void* proot;
+    alias proot this;
+}
+
+struct Range
+{
+    void* pbot;
+    void* ptop;
+    TypeInfo ti; // should be tail const, but doesn't exist for references
+    alias pbot this; // only consider pbot for relative ordering (opCmp)
+}
+
+const uint GCVERSION = 1; // increment every time we change interface
```

Comment:
No need for that, you can use `__VERSION__` to get the dmd frontend version, and we're releasing druntime w/ that as well.
Also it's our duty to keep this interface somewhat backwards compatible.